nmrfx-processor-gui/src/main/java/org/nmrfx/processor/gui/spectra/DragBindings.java

```diff
@@ -57,91 +72,45 @@ public void mouseDragDropped(final DragEvent e) {
-                        for (String item : items) {
-                            Dataset dataset1 = Dataset.getDataset(item);
-                            if (dataset1 != null) {
-                                chart.getController().addDataset(dataset1, true, false);
-                            }
-                        }
-
-                    });
-                }
+            if (dataFormatHandlers.containsKey(DataFormat.PLAIN_TEXT)) {
+                success = dataFormatHandlers.get(DataFormat.PLAIN_TEXT).handlePaste(contentString, chart);
             }
+
         } else {
-            System.out.println("no files");
+            log.info("No compatible files dragged.");
         }
         e.setDropCompleted(success);
         e.consume();
     }
 
     public void mouseDragOver(final DragEvent e) {
         final Dragboard db = e.getDragboard();
-
         List<File> files = db.getFiles();
+        boolean isAccepted = false;
         if (db.hasFiles()) {
-            if (files.size() > 0) {
-                boolean isAccepted;
-                try {
-                    isAccepted = NMRDataUtil.isFIDDir(files.get(0).getAbsolutePath()) != null;
-                    if (!isAccepted) {
-                        isAccepted = NMRDataUtil.isDatasetFile(files.get(0).getAbsolutePath()) != null;
-                    }
-                } catch (IOException ex) {
-                    isAccepted = false;
-                }
-                if (isAccepted) {
-                    // fixme
-//                    chart.setStyle("-fx-border-color: green;"
-//                            + "-fx-border-width: 1;");
-                    e.acceptTransferModes(TransferMode.COPY);
-                }
+            if (!files.isEmpty()) {
+                isAccepted = NMRDataUtil.isFIDDir(files.get(0).getAbsolutePath()) != null
+                        ||  NMRDataUtil.isDatasetFile(files.get(0).getAbsolutePath()) != null
+                        || SDFile.isSDFFile(files.get(0).getName());
             }
         } else if (db.hasString()) {
             String contentString = db.getString();
-            String[] items = contentString.split("\n");
-            if (items.length > 0) {
-                Dataset dataset = Dataset.getDataset(items[0]);
-                if (dataset != null) {
-                    // fixme
-//                    chart.setStyle("-fx-border-color: green;"
-//                            + "-fx-border-width: 1;");
-                    e.acceptTransferModes(TransferMode.COPY);
-                }
-            }
+            isAccepted = SDFile.inMolFileFormat(contentString) || Dataset.getDataset(contentString.split("\n")[0]) != null;
         } else {
             e.consume();
         }
+        if (isAccepted) {
+            // fixme
+//                    chart.setStyle("-fx-border-color: green;"
+//                            + "-fx-border-width: 1;");
+            e.acceptTransferModes(TransferMode.COPY);
+        }
```

Comment:
This fixme is from 2018, and was not related specifically to the jira issue, so did not fix, sonar lint will probably complain about this because I moved the code.

Reply:
Go ahead and remove this (chart.setStyle)